# MLB2017
## Machine Learning for Biologists
Jul 2017 - Fondazione Edmund Mach, Trento

 <table style="width:100%">
  <tr>
    <td><img src="./img/elixir_ita_logo.png" alt="yay" height="100" width="150"></td>
    <tr/>
</table>

Venue: Fondazione Edmund Mach, Trento
Date: TBD (July 2017)
Duration: 4 days
Description: The course will provide a full bioinformatic workflow for molecular profiling through preprocessing, machine learning and network visualization with R and Python.
Prerequisites: basics of R, Python, \*nix (Unix or Linux) shell. Â 
Data preparation: from GEO datasets (e.g., microarray and RNA-seq) to data matrix via \*nix shell scripting
Data exploration and preprocessing: qualitative analysis; filtering, centering and scaling of predictors
Classification basics: machine learning, model selection, and prediction
The Data Analysis Plan (DAP) - intro to unbiased pipelines for (binary) classification: theory and implementation of a basic DAP in Python (Scikit-Learn) with feature ranking and classification
Performance measures and diagnostic plots - Accuracy, MCC, Stability: theory and graphics in R (ggplot2) and Matplotlib
Differential network analysis -- co-expression networks, graph comparison, community detection: theory and examples in R/Python, visualization by the igraph library and use of the ReNette web interface.


Prerequisites: basics of R, Python, \*nix (Unix or Linux) shell. Basic statistical knowledge

Instructors: Giuseppe Jurman (Fondazione Bruno Kessler, Trento, IT), Marco Chierici (Fondazione Bruno Kessler, Trento, IT), Davide Albanese (Fondazione E. Mach, Trento, IT), Pietro Franceschi (Fondazione E. Mach, Trento, IT), Marco Moretto (Fondazione E. Mach, Trento, IT), Paolo Sonego (Fondazione E. Mach, Trento, IT), Samantha Riccadonna (Fondazione E. Mach, Trento, IT).

The course will be highly interactive and the students will continuously put theory into practice while learning. By the end of the course, the participants will ... . Basic Unix/Linux skills is a requirement.



#### Program:



Days |Morning session |Afternoon session | Trainers | Helpers|
------------ | ------------- | ------------- |
day 1 Introduction |Morning session. Introduction to "omics" data. Principles of data exploration and  analysis | Practical session with R. Univariate and Multivariate analysis|  P. Franceschi | S. Riccadonna, D. Albanese. |
day 2 Machine Learning | Machine Learning: introduction and applications to biological data. Classification basics, model selection and prediction. Performance measures and diagnostic plots |Pipelines for binary classification. Implementation of a basic analysis protocols in Python (Scikit-Learn) |G. Jurman, M. Chierici |  D. Albanese |
day 3 Analyzing Gene Expression Data |Gene Expression (microarray/RNA-Seq) data analysis: pre-processing, exploratory data analysis , differential expression analysis and visualization using R and Bioconductor| Supervised classification and feature selection of gene expression data |  P. Sonego, M. Chierici, G. Jurman | S. Riccadonna.
day 4 Advanced Machine Learning and Network Analysis| Gene prediction methods as an example of ML on genomic data|Differential network analysis -- co-expression networks, graph comparison, community detection| M. Moretto, M. Chierici, G. Jurman | A. Cestaro.|



=======


### This course was taught at:
-  [Python for Life Scientists](http://bioinformaticstraining.pythonanywhere.com/course/11/) September 26-30 2016 </br>
-   [BPBR16](http://gtpb.igc.gulbenkian.pt/bicourses/BPBR16/) July 11-15 2016<br/>
